dietstory-routing/registration/views.py

```diff
@@ -1,16 +1,35 @@
-from django.shortcuts import render
-from rest_framework import viewsets
-
-from .serializers import AccountSerializer
+from django.http import HttpResponse
+from django.utils.decorators import method_decorator
+from django.views.decorators.debug import sensitive_post_parameters
+from rest_framework import views
+from rest_framework import permissions
 from .models import Accounts
+from .validate_forms import SignupForm
+
 
 # Create your views here.
-def signup(request, username, email, password, year, month, day):
-    if request.method == "GET":
-        return username
-    elif request.method == "POST":
-        return
+class SignupView(views.APIView):
+    permission_classes = (permissions.AllowAny,)
+
+    def get(self, request, *args, **kwargs):
+        return HttpResponse("Welcome to signup page. Please enter username, email, password, and birthday.", status=200)
+
+    # @method_decorator(sensitive_post_parameters('password'))
+    def post(self, request, *args, **kwargs):
 
-class AccountsViewSet(viewsets.ModelViewSet):
-    queryset = Accounts.objects.all().order_by('name')
-    serializer_class = AccountSerializer
+        params = SignupForm(request.data)
+        print((params))
+        if params.is_valid():
+            username = params.cleaned_data.get('username')
+            accounts = Accounts.objects.filter(name=username)
+            if not accounts:
+                password = params.cleaned_data.get('password1')
+                email = params.cleaned_data.get('email')
+                birthday = params.cleaned_data.get('birthday')
+                account = Accounts(name=username, password=password, email=email, birthday=birthday)
```

Comment:
Can this fail? If it does, what happens?

What if password != params.cleaned_data.get('password1')? Are we creating an account with a non-deterministic username?